src/main/java/com/ci/ContinuousIntegrationServer.java

```diff
@@ -77,8 +78,14 @@ private void cloneRepo() {
 
     }
 
-    private void build() {
-
+    /**
+     * Builds the branch that was cloned into the target directory.
+     */
+    private void build() throws IOException, InterruptedException {
+        String path = dirPath + "/Continuous-Integration/src/main/java/com/ci/ContinuousIntegrationServer.java";
+        String[] arguments = {"javac", "-cp", "servlet-api-2.5.jar:jetty-all-7.0.2.v20100331.jar", path};
```

Comment:
I think the build function should simply run ./gradlew build in the root of the cloned repo